src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int ARRAY_SIZE = 6;
+
+    public static void main(String[] args) {
+        int i;
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Figure[] figureArray = new Figure[ARRAY_SIZE];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (i = 0; i < figureArray.length / 2; i++) {
+            figureArray[i] = figureSupplier.getRandomFigure();
+        }
+
+        while (i < figureArray.length) {
+            figureArray[i++] = new Circle(colorSupplier.getRandomColor(),3);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.